types/options.go

```diff
@@ -114,11 +120,53 @@ func loadDefaultStoreOptions() {
 
 // loadStoreOptions returns the default storage ops for containers
 func loadStoreOptions() (StoreOptions, error) {
-	storageConf, err := DefaultConfigFile()
+	baseConf, err := DefaultConfigFile()
 	if err != nil {
 		return defaultStoreOptions, err
 	}
-	return loadStoreOptionsFromConfFile(storageConf)
+
+	// Load the base config file
+	baseOptions, err := loadStoreOptionsFromConfFile(baseConf)
+	if err != nil {
+		return defaultStoreOptions, err
+	}
+
+	if _, err := os.Stat(defaultDropInConfigDir); err == nil {
+		// The directory exists, so merge the configuration from this directory
+		err = mergeConfigFromDirectory(&baseOptions, defaultDropInConfigDir)
+		if err != nil {
+			return defaultStoreOptions, err
+		}
+	} else if !os.IsNotExist(err) {
+		// There was an error other than the directory not existing
+		return defaultStoreOptions, err
+	}
+
+	return baseOptions, nil
+}
+
+func mergeConfigFromDirectory(baseOptions *StoreOptions, configDir string) error {
+	return filepath.Walk(configDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+
+		// Only consider files with .conf extension
+		if filepath.Ext(path) != ".conf" {
+			return nil
+		}
+
+		// Load drop-in options from the current file
+		err = ReloadConfigurationFile(path, baseOptions)
```

Comment:
Very quickly skimming the implementation of `ReloadConfigurationFile`, it is not designed to merge things at all; it starts with `*storeOptions = StoreOptions{}`.

Reply:
A reminder: ^^^. I can’t see how this works at all. What am I missing?

Reply:
The included unit test `TestMergeConfigFromDirectory` gives me the impression that the change is working as expected, but maybe I am missing something. Can you suggest a change to that test which can bring out the failure you are anticipating? 

The way I am looking at it is, for every drop-in config file we are initializing `*storeOptions = StoreOptions{}` and then merging it with the earlier values. But again, there is a good chance I might have overlooked something. A test case would definitely help me understand it better.

Reply:
That test is not testing _merging_ at all.

As just a starting point (I don’t understand all of the c/storage option loading code, too many functions and too many code paths and too many hooks; I’m scared of that):
```patch
diff --git a/types/options_test.go b/types/options_test.go
index bd5a70c8b..92d35dc6e 100644
--- a/types/options_test.go
+++ b/types/options_test.go
@@ -227,8 +227,7 @@ graphroot = 'temp/graph2'`,
 runroot = 'should/ignore'
 graphroot = 'should/ignore'`,
                `[storage]
-runroot = 'temp/run3'
-graphroot = 'temp/graph3'`,
+runroot = 'temp/run3'`,
                `[storage]
 runroot = 'temp/run4'
 graphroot = 'temp/graph4'`,
@@ -242,14 +241,16 @@ graphroot = 'temp/graph4'`,
 
        // Set base options
        baseOptions := StoreOptions{
-               RunRoot:   "initial/run",
-               GraphRoot: "initial/graph",
+               RunRoot:        "initial/run",
+               GraphRoot:      "initial/graph",
+               TransientStore: true,
        }
 
        // Expected results after merging configurations from only .conf files
        expectedOptions := StoreOptions{
-               RunRoot:   "temp/run3", // Last .conf file (config3.conf) read overrides earlier values
-               GraphRoot: "temp/graph3",
+               RunRoot:        "temp/run3", // Last .conf file (config3.conf) read overrides earlier values
+               GraphRoot:      "temp/graph2",
+               TransientStore: true,
        }
 
        // Run the merging function
@@ -258,7 +259,5 @@ graphroot = 'temp/graph4'`,
                t.Fatalf("Error merging config from directory: %v", err)
        }
 
-       if baseOptions.RunRoot != expectedOptions.RunRoot || baseOptions.GraphRoot != expectedOptions.GraphRoot {
-               t.Errorf("Expected RunRoot to be %q and GraphRoot to be %q, got RunRoot %q and GraphRoot %q", expectedOptions.RunRoot, expectedOptions.GraphRoot, baseOptions.RunRoot, baseOptions.GraphRoot)
-       }
+       assert.DeepEqual(t, expectedOptions, baseOptions)
 }
```

Reply:
Thanks @mtrmac . 

/hold until we fix it.